plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/DebugNativeAppEngineStandardProjectTest.java

```diff
@@ -91,12 +93,22 @@ public void testDebugLaunch() throws Exception {
     assertNotNull(testProject);
     SwtBotTestingUtilities.waitUntilMenuHasItem(
         bot, () -> bot.menu("Run").menu("Debug As"), endsWith("Debug on Server"));
-    SwtBotTestingUtilities.performAndWaitForWindowChange(
-        bot,
-        () -> {
-          bot.menu("Run").menu("Debug As").menu("1 Debug on Server").click();
-        });
 
+    bot.menu("Run").menu("Debug As").menu("1 Debug on Server").click();
+    bot.waitUntil(
+        new DefaultCondition() {
+          @Override
+          public boolean test() throws Exception {
+            return bot.shell("Debug On Server").isOpen();
+          }
+
+          @Override
+          public String getFailureMessage() {
+            return "Cannot find a shell with title 'Debug On Server'";
+          }
+        });
+    SWTBotShell debugPage = bot.shell("Debug On Server");
```

Comment:
If you are worried about this returning null, you could put an explicit assertNotNull check here with a relevant message.

Reply:
I was talking about `bot.shell("Debug On Server").isOpen()` to wait when the window is not yet open. According to the Javadoc, the method seems to never return null. I actually verified this by searching for a non-existing window with a wrong title. We should be good. I added `assertNotNull` on this line anyway.

Reply:
The `bot.shell(...).isOpen()` in `test()` would have NPEd by this point.

Reply:
Yeah, true. I'll just remove the assertion. It's clear by now `bot.shell(...)` never returns null.